django/conf/global_settings.py

```diff
@@ -204,6 +204,7 @@ def gettext_noop(s):
 EMAIL_HOST_PASSWORD = ""
 EMAIL_USE_TLS = False
 EMAIL_USE_SSL = False
+EMAIL_SSL_CAFILE = None
```

Comment:
For me, this is so rare that it's not worth adding a new setting. If someone really needs it, they can always subclass `EmailBackend` and override `ssl_context` (that's why we have it as a separate property and why it's documented in 4.2 release notes):
```python
class MyFancyEmailBackend(EmailBackend):
    @cached_property
    def ssl_context(self):
        _ssl_context = super().ssl_context
        _ssl_context.load_verify_locations(cafile="file with a custom CA")
        return _ssl_context
```

Reply:
having to extend it seems a bit excessive for most enterprises which normally have an internal CA, eventually "fixing" the functionality and allowing to set the cafile from EMAIL_SSL_CERTFILE might be better as suggested in the ticket?

Reply:
on the other hand it would be a one line long config change ;)

but if thats "too much" i would wonder why the EMAIL_SSL_KEYFILE is even implemented (which seems to mena that the email client authenticates itself against the mail server ?) but thats another topic

Reply:
> on the other hand it would be a one line long config change ;)

We have so many settings that adding a new one is always controversial (especially when there is a way to achieve the same effect).

Reply:
I agree that a new setting is not the way forward here

Instead it might be worth documenting `EmailBackend.ssl_context` in the [EmailBackend docs](https://docs.djangoproject.com/en/dev/topics/email/#smtp-backend), so that the advice on how to customize this isn't only in the Django 4.2 release notes

Reply:
> having to extend it seems a bit excessive for most enterprises which normally have an internal CA, eventually "fixing" the functionality and allowing to set the cafile from EMAIL_SSL_CERTFILE might be better as suggested in the ticket?

But wouldn't you just add the CA certificate to the system truststore then and Python/Django would pick that up automatically? The cases where you solely would need to set this for mails and not something else seem rare?

Reply:
> wouldn't you just add the CA certificate to the system truststore…?

I suggested some [doc updates along those lines](https://code.djangoproject.com/ticket/35653#comment:12:~:text=If%20we%20don%27t,point%20to%20it.) in the original ticket.

But given that [this](https://code.djangoproject.com/ticket/34504) keeps [coming](https://code.djangoproject.com/ticket/34524) up (and seems to attract "just disable SSL security" advice), I'm open to the idea of supporting it directly in the SMTP EmailBackend.

Either way, I also agree with Sarah's suggestion to document `smtp.EmailBackend.ssl_context` for overrides.

Reply:
I added some docs clarification to explicitly mention that the cert/key is for client authentication.